### PR TITLE
Feature/number input and pattern validation

### PR DESCRIPTION
**Pull request by rs-mgb**

### CODE REVIEW

File: src/es/components/organisms/registerForm/RegisterForm.js [hunk 1]
```diff
@@ -257,6 +257,109 @@ export default class RegisterForm extends Shadow() {
...
+        ) {
+          e.target.setCustomValidity("");
+          e.target.reportValidity();
+        } else {
+          e.target.setCustomValidity(validityMessage);
+          e.target.reportValidity();
+          elem.value = ""
+        }
+      })
+    })
```
Comment (cod3cow) [inline src/es/components/organisms/registerForm/RegisterForm.js]: Where do these requirements come from? I don't understand why the frontend should intercept this. Is the HTML validation using the pattern attribute not sufficient?

File: src/es/components/pages/Register.html [hunk 2]
```diff
@@ -527,6 +527,70 @@ <h2>Les Coordonnées du responsable</h2>
...
+                    <div class="form-group">
+                        <h5>Pattern validation by clicking submit button</h5>
+                        <label class="control-label" for="Data_AccountManagerLastName">Registration Number *</label>
+                        <div>
+                            <input
+                            class="form-control" 
+                            id="Data_AccountManagerLastName"
+                            name="Data.AccountManagerLastName1" 
+                            type="text" 
+                            pattern="CHE-[0-9]{3}\.[0-9]{3}\.[0-9]{3}"
```
Comment (cod3cow) [inline src/es/components/pages/Register.html:539]: LGTM

**Review by cod3cow — CHANGES_REQUESTED** [review]
Please let me know again for approval after the answer.

File: src/es/components/organisms/registerForm/RegisterForm.js (hunk 1 above)
Comment (rs-mgb) [reply src/es/components/organisms/registerForm/RegisterForm.js]: On our last meeting, when we presented our work (MigrosPro Sprint Review 09.10.2023), we decided to have a "real-time" validation on the "Number Of CustomerCard" and on the "Registration Number" input fields. Therefore we need these 2 function in JS file.
But for keeping both opportunities on the bag, i have added "sample" input field with the other kind of validation too, with pattern, maxLength, min and max HTML attributes.

Comment (cod3cow) [reply src/es/components/organisms/registerForm/RegisterForm.js]: Yes, good. Thank you for the clarification.

File: src/es/components/pages/Register.html [hunk 3]
```diff
@@ -527,6 +527,70 @@ <h2>Les Coordonnées du responsable</h2>
...
+                            class="form-control" 
+                            id="Data_AccountManagerLastName"
+                            name="Data.AccountManagerLastName2" 
+                            type="text" 
+                            value="" 
+                            maxlength="15" 
+                            custom-pattern-validation 
+                            custom-validity-message="Bitte geben Sie die richtige Format CHE-123.123.123" 
+                            placeholder="CHE-XXX.XXX.XXX" 
+                            required>
```
Comment (cod3cow) [inline src/es/components/pages/Register.html:560]: It is not possible to enter a hypen or minus (-) to confirm the input.
👇🏻 
Screenshot 2023-10-05 at 14 48 41

**Review by cod3cow — CHANGES_REQUESTED** [review]

File: src/es/components/pages/Register.html (hunk 3 above)
Comment (rs-mgb) [reply src/es/components/pages/Register.html:560]: This was a good catch and helped me to recognize an another problem with prevent defaulting some keydown events.
The problem I have recognized is the following:
On different keyboards and on different keyboard language settings, the numbers are on different keyCodes, therefore preventing keydown events based on keyCodes is not  a good idea at all.
I have removed the whole keydown listener, it means that the user is able to write now everything but after the user change the input, the validation triggers.
@ASchmidt1024 let me know what you think, thanks.

**Review by cod3cow — APPROVED** [review]
🎸 Sounds like Music!

**Pull request merged**